lib/identity/client.js

```diff
@@ -205,6 +206,62 @@ class Client extends PartialClient {
     const newSecretCreated = this.createSecret(newSecret)
     return newSecretCreated
   }
+  /**
+   * shareSecretWithUsername
+   *
+   * @param {String} secretType
+   * @param {String} secretName
+   * @param {Array} usernameToAdd
+   *
+   *
+   */
+  async shareSecretWithUsername(secretName, secretType, usernamesToAdd = []) {
+    // Generate Group to Validate it exists
+    const groupName = `tozny.secret.${this.config.realmName}.${
```

Comment:
I am not convinced that every single secret type needs its own group. Each user has a group for all their secrets and then secrets can be added to other groups. But it seems that for direct shares it makes more sense to just share with clients directly.

Reply:
I was following what is written in create Secret, but i can alter that method if we just want one group per person?